File: c/codechef/DYNAMO.cpp
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

typedef unsigned long long ulli;

int main(void)
{
        int t;
        ulli n, s, a, b, c, d, e, verdict;
        scanf("%d", &t);
        while(t--)
        {
                scanf("%llu %llu", &n, &a);
                s = 2*pow(10, n) + a;
                
                printf("%llu\n", s);
                fflush(stdout);
                
                scanf("%llu", &b);
                c = pow(10, n) - b;

                printf("%llu\n", c);
                fflush(stdout);
                
                scanf("%llu", &d);
                e = pow(10, n)-d;

                printf("%llu\n", e);
                fflush(stdout);
                
                ulli ans = a+b+c+d+e;
                printf("%llu\n", ans);
                printf("%llu\n", s);
                fflush(stdout);
                
                scanf("%llu", &verdict);
                if (verdict == -1)
                {
                        exit(0);
                }
        }
}

File: c/codechef/dynamo.py
import sys

t = int(input())

for i in range(t):
    n = int(input())
    a = int(input())
    s = 2*pow(10, n) + a
    print(s, flush=True)
    b = int(input())
    c = pow(10, n) - b
    print(c, flush=True)
    d = int(input())
    e = pow(10, n)-d
    print(e, flush=True)
    verdict = int(input())
    if verdict == -1:
        sys.exit(0)
